src/BCdpManager.t.sol

```diff
@@ -1127,4 +1169,92 @@ contract BCdpManagerTest is BCdpManagerTestBase {
 
         manager.shift(cdpSrc, cdpDst);
     }
+
+
+    function testChangePoolContract() public {
+        pool = deployNewPoolContract();
+
+        // No change in score
+        manager.setBParams(address(pool), BCdpScoreLike(address(score)));
+
+        uint cdp = manager.open("ETH", address(this));
+        reachTopup(cdp);
+    }
+
+    function testChangeScoreContract() public {
+        timeReset();
+
+        score = deployNewScoreContract();
+
+        // no change in pool contract
+        manager.setBParams(address(pool), BCdpScoreLike(address(score)));
+
+        uint cdp = manager.open("ETH", address(this));
+
+        reachTopup(cdp);
+
+        uint fwdTimeBy = 10;
+        forwardTime(fwdTimeBy);
+
+        uint expectedInkScore = 1 ether * fwdTimeBy;
+        uint exoectedArtScore = 50 ether * fwdTimeBy;
+        expectScore(cdp, "ETH", expectedInkScore, exoectedArtScore, 0);
+
+        // user ink + liquidator ink
+        uint expectedInkGlobalScore = (1 ether + 1 ether) * fwdTimeBy;
+        // uint art + liquidator art
+        uint expectedArtGlobalScore = (50 ether + 51 ether) * fwdTimeBy;
+        expectGlobalScore("ETH", expectedInkGlobalScore, expectedArtGlobalScore, 0);
+
+    }
+
+    function testChangePoolAndScoreContracts() public {
+        timeReset();
+        pool = deployNewPoolContract();
+        score = deployNewScoreContract();
+
+        manager.setBParams(address(pool), BCdpScoreLike(address(score)));
+
+        uint cdp = manager.open("ETH", address(this));
+        reachTopup(cdp);
+
+        uint fwdTimeBy = 10;
+        forwardTime(fwdTimeBy);
+
+        uint expectedInkScore = 1 ether * fwdTimeBy;
+        uint exoectedArtScore = 50 ether * fwdTimeBy;
+        expectScore(cdp, "ETH", expectedInkScore, exoectedArtScore, 0);
+
+        // user ink + liquidator ink
+        uint expectedInkGlobalScore = (1 ether + 1 ether) * fwdTimeBy;
+        // uint art + liquidator art
+        uint expectedArtGlobalScore = (50 ether + 51 ether) * fwdTimeBy;
+        expectGlobalScore("ETH", expectedInkGlobalScore, expectedArtGlobalScore, 0);
+
+    }
+
+    function testFailNonMemberTopupNewPool() public {
```

Comment:
what i meant is that non admin should try to call set b params (need to do a dedicated contract for that, or add the fuctionallity to add user). Likewise for score update.